app/code/Magento/Catalog/Pricing/Price/CalculateCustomOptionCatalogRule.php

```diff
@@ -59,7 +59,7 @@ public function execute(
             $product
         );
         // Apply catalog price rules to product options only if catalog price rules are applied to product.
-        if ($catalogRulePrice < $regularPrice) {
+        if ($catalogRulePrice < $regularPrice || $catalogRulePrice == 0.00 && $regularPrice == 0.00) {
```

Comment:
Please can you add parenthesis to make it clear that you mean `a || (b && c)` and not `(a || b) && c`.

Reply:
**Done**  parenthesis were added to condition.